daemon/api_test.go

```diff
@@ -1151,3 +1153,147 @@ func (s *apiSuite) TestInstallLicensed(c *check.C) {
 	err := inst.dispatch()()
 	c.Check(err, check.IsNil)
 }
+
+func (s *apiSuite) TestGetCapabilities(c *check.C) {
+	d := newTestDaemon()
+	d.capRepo.Add(&caps.Capability{"serial-port", "A serial port", caps.FileType, nil})
+	req, err := http.NewRequest("GET", "/1.0/capabilities", nil)
+	c.Assert(err, check.IsNil)
+	rec := httptest.NewRecorder()
+	capabilitiesCmd.GET(capabilitiesCmd, req).ServeHTTP(rec, req)
+	c.Check(rec.Code, check.Equals, 200)
+	c.Check(rec.Body.String(), testutil.Contains, "serial-port")
+	c.Check(rec.Body.String(), check.Equals,
+		`{"result":[{"name":"serial-port","label":"A serial port",`+
+			`"type_name":"file","attrs":null}],`+
```

Comment:
checking json string for equals is very fragile, as for a start there is no order guarantee for the attributes of an object, and to go on whitespace is not significant.

About omitempty, agreed.